add support for Ad5940

buha opened the pull request:
# Description

add support for ad5940 body impedance analysis (impedance reading)

Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] examples/ad5940.py
- [x] sw_eit_ad5940 app, modified to run using pyadi-iio

**Test Configuration**:
* Hardware: cn0565
* OS: linux

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

tfcollins reviewed (changes requested):
Please add some basic verification tests for the attributes and RX DMA

buha commented:
- added attribute tests

This chip doesn't propose a buffered IIO interface so no need to add DMA tests.

buha commented:
codespell is complaining about spelling "byteorder" instead of "byte order", i guess it's a false alarm

perhaps there is a way to disable codespell inline ? i couldn't find any in a quick search

tfcollins commented:
Linting errors:
```
codespell................................................................Failed
- hook id: codespell
- exit code: 4
adi/ad5940.py:139: byteorder  ==> byte order  | disabled due to java class
adi/ad5940.py:144: byteorder  ==> byte order  | disabled due to java class
adi/ad5940.py:147: byteorder  ==> byte order  | disabled due to java class
adi/ad5940.py:153: byteorder  ==> byte order  | disabled due to java class
isort....................................................................Failed
- hook id: isort
- files were modified by this hook
Fixing /home/runner/work/pyadi-iio/pyadi-iio/adi/__init__.py
```

buha commented:
Ok did some research and saw there is a codespell whitelist.
Added an extra commit to put `byteorder` in the codespell whitelist.

Rebased and fixed pre-commit.
